internal/utils/events.go

```diff
@@ -0,0 +1,117 @@
+package utils
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/Taraxa-project/taraxa-indexer/models"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// LogReward ..
+type LogReward struct {
+	Account   string
+	Validator string
+	Value     *big.Int
+	EventName string
+}
+
+type RewardsClaimedEvent struct {
+	Account   string
+	Validator string
+	Amount    *big.Int
+}
+
+type CommissionRewardsClaimedEvent struct {
+	Account   string
+	Validator string
+	Amount    *big.Int
+}
+
+const dposABI = `[{"anonymous":false,"inputs":[{"indexed":true,"internalType":"address","name":"account","type":"address"},{"indexed":true,"internalType":"address","name":"validator","type":"address"},{"indexed":false,"internalType":"uint256","name":"amount","type":"uint256"}],"name":"RewardsClaimed","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"address","name":"account","type":"address"},{"indexed":true,"internalType":"address","name":"validator","type":"address"},{"indexed":false,"internalType":"uint256","name":"amount","type":"uint256"}],"name":"CommissionRewardsClaimed","type":"event"},{"inputs":[{"internalType":"address","name":"validator","type":"address"}],"name":"claimCommissionRewards","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"validator","type":"address"}],"name":"claimRewards","outputs":[],"stateMutability":"nonpayable","type":"function"}]`
```

Comment:
I am adding our dpos_client to indexer that should have ABI somewhere inside of it. Maybe you should also get it from there to have it in one place 
https://github.com/Taraxa-project/taraxa-indexer/pull/89/files#diff-0819f00812e2733d6c042d259dbf33efedcb3042c3f3e615ee212eafd89bb00eR9

Reply:
For some reason if I import it form here the event decoding fails. ❓